functions

```diff
@@ -29,10 +29,7 @@ emergency_shell() {
 
 detect_virt() {
    # Detect LXC containers
-   [ ! -e /proc/self/environ ] && return
-   if grep -q lxc /proc/self/environ >/dev/null; then
-       export VIRTUALIZATION=1
-   fi
+   test -z "$container" || export VIRTUALIZATION=1
```

Comment:
This only checks whether the variable is not empty, not whether it is set.
I'd recommend using the (deleted) approach of grepping:

```bash
grep -zq ^container= /proc/self/environ 2>/dev/null && export VIRTUALIZATION=1
```

Reply:
not gonna work with busybox

```
/ # busybox grep -z
grep: invalid option -- 'z'
```

Reply:
On one hand I doubt so, on the other hand I've seen enough not to be particularly inclined to trust that assumption.
Anyways, which version of busybox are you using, the one I tested on my voidlinux installation and the one on my Gentoo box both have a `-z` flag.
The Gentoo box running BusyBox v1.31.1.

Reply:
```
cameronnemo@cn-mbook ~> docker run --rm -it voidlinux/voidlinux sh
# xbps-install -Syu ; xbps-install -yu ; xbps-install -y busybox
[...]
# busybox grep -zq ^container= /proc/self/environ
grep: invalid option -- 'z'
BusyBox v1.30.1 (2019-08-02 18:59:22 UTC) multi-call binary.

Usage: grep [-HhnlLoqvsriwFE] [-m N] [-A/B/C N] PATTERN/-e PATTERN.../-f FILE [FILE]...
```

Reply:
What's the purpose of this? Why not use `[ "${container+x}" ]`

Reply:
@leahneukirchen entirely forgot about that construct.
Right, that seems more reasonable.

Reply:
changed to use the recommended construct. Thanks @leahneukirchen.